Sync/RustSyncManagerAPI.swift

```diff
@@ -0,0 +1,81 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/
+
+import Foundation
+import Logger
+import Shared
+@_exported import MozillaAppServices
+
+open class RustSyncManagerAPI {
+    private let logger: Logger
+    var api: SyncManager
+
+    public init(logger: Logger = DefaultLogger.shared) {
+        self.api = SyncManager()
+        self.logger = logger
+    }
+
+    public func disconnect() {
+        DispatchQueue.global(qos: .default).async {
+            self.api.disconnect()
+        }
+    }
+
+    public func sync(
+        params: SyncParams,
+        completion: @escaping (MozillaAppServices.SyncResult) -> Void
+    ) {
+        DispatchQueue.global(qos: .default).async {
```

Comment:
Curious if there's any reasoning behind defining the dispatch queue at this layer

There are a few ways to go about this, and I'm genuinely unsure how we want to handle it:
- We can keep it here, and use the global default queue like what we have here, it's fine, but the sync will now compete with other default queue operations.. this means it's possible if a sync takes too long it will delay other tasks, and vice versa (which may be okay?)
- We can define our own dispatch queue so we don't compete with other queues, but I imagine we'd want to justify it so getting iOS engineers' opinions before we do this is probably a good call
- We can keep it synchronous and have the caller make sure it's running in an async queue. The benefit here is that if we do a background sync, the caller can have it in the background queue, etc


Thoughts? I'm leaning towards the second, to keep syncs on their own queue, but I'm not sure if iOS has a policy against that (which could be reasonable!)

Reply:
I actually talked with Nish about this last week because I wasn't sure what the best option was either. He suggested using `DispatchQueue.global` here.  But now I'm actually leaning towards having the queues in the callers. @nbhasin2 Do you have a recommendation here?

Reply:
It depends on the scenario, do these operations depend on anything in this class or are other operations dependent on this? 

1. You don't need `DispatchQueue.global(qos: .default).async` as that is same as `DispatchQueue.global().async` :) 
2. If you want to have async calls but on one of the background threads then use the following 
```
DispatchQueue.global(qos: .background).async {
// put code here
}
```
3. If you want to batch bunch of calls into a single queue then it depends on sync vs asyc. But for async concurrent queue the following works
```
let syncmanagerQueue = DispatchQueue(label: "syncmanagerqueue", attributes: .concurrent)
syncmanagerQueue.sync {
// put code here
}  
```

I would suggest `1` if you don't have any dependency and can handle this operation at any time of the lifecycle of the app.